doc/sphinx/src/using-closures.rst

```diff
@@ -3,131 +3,362 @@
 
-The average density in a given material-occupied volume is
+where each material has its own density, :math:`\rho_i`, and specific internal
+energy, :math:`e_i`, as well as in principle its own pressure and temperature.
+Each will be a function of the bulk density, :math:`\rho`, the bulk specific
+internal energy, :math:`\epsilon`, and both the mass fraction, :math:`\mu_j`,
+of all materials present.
+
+For convenience of minimizing input arguments though, the solvers in
+``singularity-eos`` are typically posed in a different way that can sometimes be
+unintuitive.
+
+For some finite total volume :math:`V`, each material occupies some fraction of
+that volume given by the *volume fraction*
+:math:`f_i` such that
 
 .. math::
 
-  \bar{\rho}_i = \rho_i f_i
+  \sum_{i=0}^{N - 1} f_i = f_\mathrm{tot},
+
+where :math:`f_\mathrm{tot}` is the total fraction of the total volume being
+considered (in principle, different closure models can be used for different
+sets of materials). To consider the entire volume, :math:`f_\mathrm{tot}` can
+simply be set to one.
+
+The average density, :math:`\overline{\rho}_i`, (i.e. mass per *total* volume)
+for a material in the total volume is
+
+.. math::
+
+  \overline{\rho}_i = \rho_i f_i,
+
+where :math:`\rho_i` is the physical density (i.e. material mass per *material*
+volume). The total density (mass of *participating* materials per total volume)
+is then
+
+.. math::
+
+  \rho = \sum_{i=0}^{N - 1} \overline{\rho}_i = \sum_{i=0}^{N-1} \rho_i f_i
+
+Similarly the energy can be summed in a similar way so that
+
+.. math::
+
+  u = \rho \epsilon = \sum_{i = 0}^{N - 1} \rho_i \epsilon_i 
+  = \sum_{i = 0}^{N - 1} u_i
+
+where :math:`u` is the total internal energy density (internal energy per unit
+volume). Similarly, :math:`u_i` is analagous to :math:`\overline{\rho}_i` in
+that it is the internal energy for a material averaged over the entire control
+volume.
+
+Internally, the closer models in ``singularity-eos`` operate on :math:`f_i`,
+:math:`\overline{\rho}_i`, and :math:`u_i` as well as their total counterparts.
+ This is different than the forms stated at the beginning of this section so
+ that in essence the PTE solver has the form
+
+.. math::
+
+  P_i = F(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  \rho_i = G(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  \epsilon_i = H(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1}) \\
+  T_i = J(\epsilon, f_\mathrm{tot}, f_1, ..., f_i, ..., f_{N-1},
+          \rho_1, ..., \rho_i, ..., \rho_{N-1})
+
+
+The important nuance here is the **the volume fractions and densities are both
+inputs and outputs** in the current ``singularity-eos`` formulation of the
+closure models. From physics perspective this can be confusing, but from a code
```

Comment:
If you insist on these input variables you have to add how they are dependent on each other, that is that \overline{rho} is constant. It cannot be repeated enough.

Reply:
I repeated the total density equation in the warning block to re-emphasize this point

Reply:
Maybe you can discuss this at the point you first talk about \overline{bar}? Pointing out that holding it constant means that the density is constant but that both rho_i and f_i can change as long as their product is constant.

Reply:
The density equation itself doesn't say that \overline{bar} is constant since only the sum of all of them need to be constant.

Reply:
I'll repeat it there as well